### PR TITLE
SVG nodes (WIP)

### PR DESCRIPTION
**Pull request by vicdoval**
Scalable  Vector Graphics Nodes:

image

- [ ] Code changes complete.
- [ ] Code documentation complete.
- [ ] Documentation for users complete (or not required, if user never sees these changes).
- [ ] Manual testing done. 
- [ ] Unit-tests implemented.
- [ ] Ready for merge.

### CODE REVIEW

File: nodes/svg/text.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    def draw(self, height, scale):
+        svg = '<text '
+
+        svg += 'font-size="{}px" '.format(self.size)
+        svg += 'font-family="{}" '.format(self.font_family)
+        x = self.location[0] * scale
+        y = height-self.location[1] * scale
+        if self.angle != 0:
+            svg += 'transform="translate({},{})rotate({})" '.format(x,y,self.angle)
```
Comment (zeffii) [inline nodes/svg/text.py:48]: you don't like `f string` formatting?
```
svg += 'transform="translate({},{})rotate({})" '.format(x,y,self.angle)
svg += f'transform="translate({x},{y}) rotate({self.angle})" '
```

Comment (zeffii) [reply nodes/svg/text.py:48]: either way. super cool stuff.

**Comment by portnov** [comment]
Eeh :)
What do these nodes do? Generate an SVG file? or generate some geometry in Blender / Sverchok?

**Comment by zeffii** [comment]
would be neat to spawn a localhost, pointing at that .svg for a live response.

**Comment by vicdoval** [comment]
@portnov  there was a node that fade in from a previous failed experiment (solid from loft) but this PR is about generating SVG files from Sverchok

File: nodes/svg/text.py (hunk 1 above)
Comment (vicdoval) [reply nodes/svg/text.py:48]: @zeffii Thanks! i didn't know the f'string formatting but looks super neat and i will move forward to using it

**Comment by portnov** [comment]
Hmm. @vicdoval @zeffii I'm not sure it's a good idea to have such nodes in sverchok... May be it's better to put them into separate addon? We do not put audio or texturing nodes into sverchok, after all...

**Comment by vicdoval** [comment]
> would be neat to spawn a localhost, pointing at that .svg for a live response.

That would be ultra-cool but i don't know how to implement it (and how responsive it would be). Any help would be appreciated 😄

**Comment by vicdoval** [comment]
@portnov the basic idea is to be able to produce blueprints, technical drawings or files for CNC production from Sverchok. I needed that feature in the past and I did it with some SNL scripts. At the beginning I thought of a single node to do it but I found more convenient to split it in some smaller nodes easier to maintain and combine.
Of course it opens many other possibilities that lead to the idea of a svechok-extra addon but that has another troubles that would need to be fixed (addition to examples, searching from menu, tests, addon-github-update)

image

File: docs/nodes/svg/svg_index.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+*****
+Solid
```
Comment (zeffii) [inline docs/nodes/svg/svg_index.rst:2]: svg

**Comment by vicdoval** [comment]
Seems that Travis is on holidays (at least from the last 5 days) 😄

**Comment by zeffii** [comment]
i've got a nice csv that'll test this on :)

**Comment by vicdoval** [comment]
I would appreciate an update in the docs or even better to establish a automatic update system of the documentation

**Comment by nortikin** [comment]
> I would appreciate an update in the docs or even better to establish a automatic update system of the documentation

yes, it will be better. Maybe to publicate on github.io automatically?

**Pull request merged**